### ISSUE TITLE
Frontend for enabling MFA in site settings

### ISSUE THREAD

**Issue by brynwhyman** [Firesphere/silverstripe-bootstrapmfa#24]
### User story
As a CMS admin, I want the ability to enable MFA controls from within the site settings, so that I can enable this in time with my organisation's MFA implementation plan

### ACs (front-end only)
- [ ] There is a checkbox for enabling/disabling MFA for a site with the label "Enable MFA for CMS access"
- [ ] Ticking the checkbox enables the below radio buttons, else they are read-only
- [ ] Two radio buttons allow for either:
 - non-mandatory MFA, with the label "MFA is optional for everyone"
 - mandatory MFA, with the label "MFA is required for everyone"
- [ ] "MFA is optional for everyone" is the default if the checkbox is ticked

**Comment by Firesphere** [comment]
> There is a checkbox for enabling/disabling MFA for a site with the label "Enable MFA for CMS access"

This is a strange wording, as with the installation of a module, it becomes available.
I'd imagine this feature to be related to the showing of the information during login, where, once ticked, the information shows to the user that it should preferably enable the MFA. Until the checkbox is ticked, MFA is available, but not actively advertised to the CMS Users.

**Comment by brynwhyman** [comment]
The above AC was made to cater for a site that's had MFA installed before the users are ready to use it (based on lack of communications/training by site admin), giving the admin the power to enable it when the organisation is ready. As MFA becomes more prevalent for online services I expect this scenario will become less frequent.

We could alter the 'Enable MFA for CMS access' checkbox to either enable/disable the MFA setup prompt after login, as @Firesphere suggested above. OR, just remove this and have 'MFA is optional for everyone' be the default when MFA is installed on a site, with the only alternative being it's mandatory.
![image](https://user-images.githubusercontent.com/15868440/47622082-969f4400-db65-11e8-9b25-18c775409edd.png)

There's been some iterations with this functionality. Keen on your thoughts @clarkepaul @newleeland

**Comment by ScopeyNZ** [comment]
I tend to agree with @Firesphere . I don't think we should have an "Enable MFA" checkbox. This should just be available if installed and correctly configured. An educated power user can opt in if they wish - pre-empting any eventual requirement for all users.

MFA optional for everyone should just be "MFA is optional but a prompt is shown on login" or whatever.

**Comment by clarkepaul** [comment]
Like @brynwhyman mentioned it was to cater to easier role-out for agencies, specially if there are multiple sites or cross-agency logins, or transitioning to/from other authentication services like RealMe (they might want to role out MFA across different sites at the same time). I feel that providing the control to the admin is pretty empowering, and they will be able to plan the rollout to their own timeframe and not dictated by a dev agency.

Agree that if time is short this is something that could drop although for the amount of effort required I see as being beneficial.

@Firesphere actually Sub-sites probably needs a bit of discussion, is everything based off the main site or does each site have its own MFA settings, any thoughts?

**Comment by Firesphere** [comment]
> @Firesphere actually Sub-sites probably needs a bit of discussion, is everything based off the main site or does each site have its own MFA settings, any thoughts?

I've not taken subsites into account, but if the user (Which is global) has MFA set up, it's mandatory automatically. The grace period per subsite is based on the same principal from SiteConfig

**Comment by clarkepaul** [comment]
Cool that was my expectation. @brynwhyman I've created a story to capture that MFA works with Subsites—MFA settings are managed on the main site and this flows through for the Subsites. See https://github.com/Firesphere/silverstripe-bootstrapmfa/issues/37
